docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,327 @@
+
+As mentioned, `controlledValues` influences whether only requests or requests and limits are scaled:
+- `RequestsOnly`: Updates only requests and doesn't change limits. Useful if you have defined absolute limits (unrelated to the requests).
+- `RequestsAndLimits` (default): Updates requests and proportionally scales limits along with the requests. Useful if you have defined relative limits (related to the requests). In this case, the gap between requests and limits should be either zero for QoS `Guaranteed` or small for QoS `Burstable` to avoid useless (way beyond the threshold of unhealthy behavior) or absurd (larger than node capacity) values.
+
+VPA doesn't offer many more settings that can be tuned per VPA resource than you see above (different than HPA's `behavior` section). However, there is one more that isn't shown above, which allows to [scale only up or only down (`evictionRequirements[].changeRequirement`)](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/README.md#controlling-eviction-behavior-based-on-scaling-direction-and-resource), in case you need that, e.g. to provide resources when needed, but avoid disruptions otherwise.
+
+## VPA Options
+
+VPA is an independent community project that [consists of](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/README.md#components-of-vpa) a recommender (computing target recommendations and bounds), an updater (evicting pods that are out of recommendation bounds), and an admission controller (mutating webhook applying the target recommendation to newly created pods). As such, they have independent options.
+
+### VPA Recommender Options
+
+You can read up the [full VPA recommender options](https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/FAQ.md#what-are-the-parameters-to-vpa-recommender) online and set some of them conveniently in your [Gardener shoot cluster spec](https://github.com/gardener/gardener/blob/957e4c7/example/90-shoot.yaml#L298-L307):
+- `recommendationMarginFraction` (default 15%): Safety margin that will be added to the recommended requests.
+- `targetCPUPercentile` (default 90%): CPU usage percentile that will be targeted with the CPU recommendation. This setting is relevant for balancing your availability needs vs. your costs. Reducing this value is possible, because CPU is compressible, but then VPA may lack the necessary signals for scale-up as throttling on an otherwise fully utilized node will go unnoticed by VPA.
+- `targetMemoryPercentile` (default 90%): Memory usage percentile that will be targeted with the memory recommendation. This setting is relevant for balancing your availability needs vs. your costs. Reducing this value is possible, because OOMs will trigger bump-ups, but those will disrupt the workload.
```

Comment:
It is a knob you can turn: If you target a higher and higher percentile (up to 100%) you plan more and more for the worst case -> higher likelihood of over-reservation, but also higher costs. Conversely, if you target a lower and lower percentile -> higher likelihood of under-reservation, but also lower costs.

Reply:
I added in both cases in brackets:
> "recommendation will "fit" e.g. 90% of the observed CPU resp. memory usages"

Reply:
It would be confusing for end users (as it is about multiple recommenders in parallel), but FYI:
https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/README.md#starting-multiple-recommenders gives examples with frugal, standard, and performance. 90% is standard. The lower you go, the less costs you will have (and higher risks) and the higher you go, the safer your recommendation will be (and higher costs).

Reply:
I also added:
> If you want to err on the safe side, you can increase this value, but you will then target more and more a worst case scenario, quickly (maybe even exponentially) increasing the costs.